reference/client-access/python-sdk.md

```diff
@@ -0,0 +1,287 @@
+and the second (`args`) being an array of values to bind to the placeholders.
+
+```py
+rs = client.execute(
+    stmt="select score from example_scores where uid = ? and level = ?",
+    args=[ "uid1", 2 ]
+)
+
+# rs.columns == ('score',)
+# rs.rows[0] == (95,)
+```
+
+### Named placeholders
+
+libSQL supports positional placeholders using the same syntax as SQLite. Pass
+two parameters, with the first (`stmt`) must be the SQL string with
+placeholders, and the second (`args`) must be a dictionary whose keys match the
+names of the placeholders, and whose values will be bound to the placeholders.
+
+```py
+rs = client.execute(
+    stmt="insert into example_scores values (:uid, :level, :score)",
+    args={ "uid": "uid2", "level": 1, "score": 50 }
+)
+
+# rs.columns == ()
+# rs.rows == []
+# rs.rows_affected == 1
+```
+
+In the above example, there are placeholders for `uid`, `level`, and `score`
+using the prefix character `:`. The values of the matching properties of `args`
+are bound to each placeholder.
+
+libSQL supports the same named placeholder prefix characters as SQLite: `:`,
+`@`, and `$`.
+
+## Execute a batch of statements
+
+:::info
+
+Be sure to read the [common section on batches] for libSQL clients to understand
+their behavior.
+
+:::
+
+To execute multiple statements in a transaction, use the `batch()` method on the
+client object, passing it an array of statements. The array may contain any type
+of statement that is also accepted by
+[`execute()`](#execute-a-single-statement). `batch()` an array of
+[ResultSet](#resultset) objects (one for each statement).
+
+The following code inserts a row for uid3 in two different tables using a
+transaction that commits them both at the same time.
+
+```py
+rss = client.batch([
+    libsql_client.Statement(
+        sql="insert into example_users values (?, ?)",
+        args=[ "uid3", "uid3@turso.tech" ]
+    ),
+    libsql_client.Statement(
+        sql="insert into example_scores values (?, ?, ?)",
+        args=[ "uid3", 1, 200 ]
+    ),
+])
+
+# rss[0].columns == ()
+# rss[0].rows == []
+# rss[0].rows_affected == 1
+# rss[1].columns == ()
+# rss[1].rows == []
+# rss[1].rows_affected == 1
+```
+
+## Interactive transactions
+
+:::info
+
+Be sure to read the [common section on interactive transactions] for libSQL
+clients to understand their behavior.
+
+:::
+
+Use the `transaction()` method on the client object to create a Transaction
+object to control the transaction. It provides the following methods:
+
+| Method | Description |
+| --- | --- |
+| `execute()` | Similar to the normal client `execute()`, except within the context of the transaction |
+| `commit()` | Commits all the write statements in the transaction |
+| `rollback()` | Rolls back the entire transaction |
+| `close()` | Immediately stops the transaction - must be called if the transaction was not committed or rolled back in order to free resources |
```

Comment:
It is worth noting that the `close()` method is sync even for the async client.

Reply:
We should also mention that the user can use a `with` statement to make sure that the transaction is closed.

Reply:
Done, and also updated the code sample to use `with`.